Source/Diagnostics/ReducedDiags/ParticleExtrema.cpp

```diff
@@ -0,0 +1,419 @@
+            chimin.resize(level_number+1,0.0_rt);
+            chimax.resize(level_number+1,0.0_rt);
+
+            // loop over refinement levels
+            for (int lev = 0; lev <= level_number; ++lev)
+            {
+                // Loop over boxes
+                for (WarpXParIter pti(myspc, lev); pti.isValid(); ++pti)
+                {
+                    // declare position arrays
+                    Gpu::ManagedDeviceVector<ParticleReal> x, y, z;
+                    // get particle arrays
+                    pti.GetPosition(x, y, z);
+                    auto & w  = pti.GetAttribs(PIdx::w);
+                    auto & ux = pti.GetAttribs(PIdx::ux);
+                    auto & uy = pti.GetAttribs(PIdx::uy);
+                    auto & uz = pti.GetAttribs(PIdx::uz);
+                    // declare external fields
+                    int offset = 0;
+                    const auto getExternalE = GetExternalEField(pti, offset);
+                    const auto getExternalB = GetExternalBField(pti, offset);
+                    // declare reduce_op
+                    ReduceOps<ReduceOpMin, ReduceOpMax> reduce_op;
+                    ReduceData<Real, Real> reduce_data(reduce_op);
+                    using ReduceTuple = typename decltype(reduce_data)::Type;
+                    reduce_op.eval(ux.size(), reduce_data,
+                    [=] AMREX_GPU_DEVICE (int i) -> ReduceTuple
+                    {
+                        // get external fields
+                        ParticleReal ex = 0._rt, ey = 0._rt, ez = 0._rt;
+                        ParticleReal bx = 0._rt, by = 0._rt, bz = 0._rt;
+                        getExternalE(i, ex, ey, ez);
+                        getExternalB(i, bx, by, bz);
+                        // gather E and B
+                        Array4<const amrex::Real> ex_arr, ey_arr, ez_arr;
+                        Array4<const amrex::Real> bx_arr, by_arr, bz_arr;
+                        IndexType ex_type, ey_type, ez_type;
+                        IndexType bx_type, by_type, bz_type;
+                        GpuArray<amrex::Real, 3> dx_arr, xyzmin_arr;
+                        Dim3 lo;
+                        int n_rz_azimuthal_modes;
+                        int nox;
+                        bool galerkin_interpolation;
+                        doGatherShapeN(x[i], y[i], z[i],
+                            ex, ey, ez, bx, by, bz,
+                            ex_arr, ey_arr, ez_arr, bx_arr, by_arr, bz_arr,
+                            ex_type, ey_type, ez_type,
+                            bx_type, by_type, bz_type,
+                            dx_arr, xyzmin_arr, lo,
+                            n_rz_azimuthal_modes, nox, galerkin_interpolation);
+                        // compute chi
+                        Real chi = 0.0_rt;
+                        if ( myspc.AmIA<PhysicalSpecies::photon>() ) {
+                            chi = chi_photon(ux[i]*m, uy[i]*m, uz[i]*m,
```

Comment:
There is actually an issue with this line: `m` for photons is 0! `ux`, `uy` and `uz` for photons are calculated as if they had the same mass of an electron. This should probably be better emphasized in the developers documentation.
Since only photons can have this QED process, you can simply replace m with the electron mass (and maybe add a one line comment)